### DISCUSSION TITLE
Write/Read binary

### DISCUSSION THREAD

**Discussion by ghost** [tauri-apps/tauri#5673] [Q&A] [answered]
Is this correct way of saving/loading binary?
```ts
const data = "Hello World!"
const binary = btoa(data)
await writeBinaryFile("...", binary)
```
```ts
const binary = await readBinaryFile("...")
const data = atob(binary)
```

**Comment by FabianLars** [comment, marked as answer]
`btoa` creates a base64 encoded _string_ in which case you'd still use writeTextFile instead.

writeBinaryFile expects actual binary data (for example image data) in form of a binary (u8) array (`Blob`s don't work)

**Reply by ghost** [reply]
Thanks!

**Reply by jswhisperer** [reply]
I think Rust FS and Event DragDrop don't include "type" by default? So you have to guess by file extension. Let me know if I'm 
missing something

Native Drag event from browser provides:
![Screenshot 2024-09-16 at 5 42 56 PM](https://github.com/user-attachments/assets/ae005cf4-e23e-433b-a0b1-ab3474b41b12)

as per every article I can find they manually add mime type https://viralsfire.com/post/errors-decoding-a-binary-image-file-using-blob-and-window

![Screenshot 2024-09-18 at 9 37 53 PM](https://github.com/user-attachments/assets/740f1052-32a8-4b81-be53-71ac00cf7ecc)


In browser webview land you can 

```
const convertBlobToBase64 = (blob) => new Promise((resolve, reject) => {
		const reader = new FileReader();
		reader.onerror = reject;
		reader.onload = () => {
			resolve(reader.result);
		};
		reader.readAsDataURL(blob);
	});
```

```
async function getClipboardContents(e) {
		e.preventDefault();

		try {
			const clipboardItems = await navigator.clipboard.read();

			for (const clipboardItem of clipboardItems) {
				for (const type of clipboardItem.types) {
					const blob = await clipboardItem.getType(type);
					console.log({ blob });
				}
			}
		} catch (err) {
			console.error(err.name, err.message);
		}
	}
```

node.fs also provides this

**Reply by jswhisperer** [reply]
I've even tried parsing the OBJECTURL, Buffer, and typed array from Tauri core methods and its missing so a lib like https://dev.to/victrexx2002/how-to-get-the-mime-type-of-a-file-in-nodejs-p6c doesn't work
